
load @textsurf/snow/amazonaws.com/appsync/2017-07-25/base/bad-request-exception
  take form bad-request-exception

load @textsurf/snow/amazonaws.com/appsync/2017-07-25/base/concurrent-modification-exception
  take form concurrent-modification-exception

load @textsurf/snow/amazonaws.com/appsync/2017-07-25/base/internal-failure-exception
  take form internal-failure-exception

load @textsurf/snow/amazonaws.com/appsync/2017-07-25/base/list-types-response
  take form list-types-response

load @textsurf/snow/amazonaws.com/appsync/2017-07-25/base/not-found-exception
  take form not-found-exception

load @textsurf/snow/amazonaws.com/appsync/2017-07-25/base/unauthorized-exception
  take form unauthorized-exception

load @textsurf/snow/base/native-integer
  take form native-integer

load @textsurf/snow/base/native-string
  take form native-string

boot list-types, name <ListTypes>
  deed get
  note <Lists the types for a given API.>

  take api-id
    like native-string
    note <The API ID.>
  take format
    like native-string
    note <The type format: SDL or JSON.>
  take next-token
    like native-string
    note <An identifier that was returned from the previous call to this operation, which can be used to return the next set of items in the list.>
    void take
  take max-results
    like native-integer
    note <The maximum number of results you want the request to return.>
    void take

  line </v1/apis/{api-id}/types#format>

  find <format>, loan format
  find <nextToken>, loan next-token
  find <maxResults>, loan max-results

  loot 200
    seed json, like list-types-response
    note <Success>
  loot 480
    seed json, like bad-request-exception
    note <BadRequestException>
  loot 481
    seed json, like concurrent-modification-exception
    note <ConcurrentModificationException>
  loot 482
    seed json, like not-found-exception
    note <NotFoundException>
  loot 483
    seed json, like unauthorized-exception
    note <UnauthorizedException>
  loot 484
    seed json, like internal-failure-exception
    note <InternalFailureException>